src/Containers/ProductDetail.js

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import axios from 'axios';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectedProduct, removeSelectedProduct } from '../Redux/Actions/productActions';
+import '../Styles/ProductDetail.css';
+
+const ProductDetail = () => {
+  const product = useSelector((state) => state.product);
+  const {
+    image, title, price, category, description, rating,
+  } = product;
+  const { productid } = useParams();
+  const dispatch = useDispatch();
+
+  const fetchProductDetail = useCallback(async () => {
+    const response = await axios.get(`https://fakestoreapi.com/products/${productid}`).catch((err) => {
+      throw new Error('Err', err);
+    });
+    dispatch(selectedProduct(response.data));
+  }, [productid, dispatch]);
+
+  useEffect(() => {
+    if (productid && productid !== '') {
+      fetchProductDetail();
+    }
+
+    return () => {
+      dispatch(removeSelectedProduct());
+    };
+  }, [productid, fetchProductDetail, dispatch]); // Include dispatch in the dependency array
+
+  return (
+    <div className="DetailsPage">
+      {Object.keys(product).length === 0 ? (
+        <div><h3>...loading</h3></div>
+      ) : (
+        <>
+          <div className="Home-header">
+            <div className="image">
+              <img className="image" src={image} alt={title} />
+            </div>
+            <h2>
+              {title}
+            </h2>
+          </div>
+          <div className="productDetail">
+            <div className="productDescription">
+              <div className="productDescription">
+                Category:
+                {category}
+              </div>
+              <div className="productDescription">
+                Product Description:
+                {description}
+              </div>
+              <div className="productDescription">
+                Price:
+                $
+                {price}
+              </div>
+              <div className="productDescription">
+                Count:
+                {rating.count}
+              </div>
+              <div className="productDescription">
+                Rating:
```

Comment:
- [ ] I highly recommend fixing the element position here so the price in the left and the number to the right since the element should be like the layout as mentioned in the requirements and kindly as mentioned by the previous reviewer, Let's ensure that the layout/positioning of elements and color pattern closely matches the template design 🔥💯✨💪🏻


- Your design
![image](https://github.com/Dorcas126/capstone-project-metrics-webapp/assets/58553711/0cb81f12-7eb7-44c3-98a3-894c11883f62)


- Given
![image](https://github.com/Dorcas126/capstone-project-metrics-webapp/assets/58553711/deed16f5-fc1f-4c17-9486-0b24299008b9)